metals/src/main/scala/scala/meta/internal/metals/formatting/MultilineString.scala

```diff
@@ -1,22 +1,26 @@
 package scala.meta.internal.metals.formatting
 
+import scala.annotation.tailrec
+
 import scala.meta.internal.metals.UserConfiguration
 import scala.meta.internal.mtags.MtagsEnrichments._
 import scala.meta.tokens.Token
-import scala.meta.tokens.Token.Interpolation
 import scala.meta.tokens.Tokens
 
 import org.eclipse.lsp4j.Position
 import org.eclipse.lsp4j.Range
 import org.eclipse.lsp4j.TextEdit
 
+private case class TokenData(startToken: Token, endToken: Token, endIndex: Int)
```

Comment:
What do you think about turning it into smth like the following and perform related refactoring?
```scala
case class StringLiteralExpr(start: Position, end: Position, hasStripMargin: Boolean)
```
`TokenData` is too wide. It took me a while to get the actual purpose of this class :smile: 
Also, `endIndex` is used only for passing it into in `hasStripMarginSuffix` and as this method is always called after `getTokenData` so it might incorporate this logic.